src/coreclr/vm/i386/PInvokeStubs.asm

```diff
@@ -49,6 +53,15 @@ _JIT_PInvokeBegin@4 PROC public
         mov             eax, [esp]
         mov             dword ptr [ecx + InlinedCallFrame__m_pCallerReturnAddress], eax
 
+ifdef FEATURE_EH_FUNCLETS
+        ;; Link SEH exception registration
+        mov             eax, dword ptr fs:[0]
+        mov             dword ptr [ecx + InlinedCallFrame__m_ExceptionRecord], eax
+        mov             dword ptr [ecx + InlinedCallFrame__m_ExceptionRecord + 4], _ProcessCLRException
+        lea             eax, [ecx + InlinedCallFrame__m_ExceptionRecord]
+        mov             dword ptr fs:[0], eax
+endif
```

Comment:
Turns out we may not need this afterall. I gave this a second thought after some inquisitive questions about the SEH overhead, how it works in the current runtime, and the discussion around enter/exit frames in an unrelated interpreter PR. We should just need one SEH registration for each managed part of the call stack and it shouldn't matter whether it's at the entry or exit point. Earlier I had some code dependent on the SEH frame being at the exit point but that's no longer the case. The current x86 EH errects the SEH frames at the entrypoint (eg. JIT_ReversePInvokeEnter). We should be able to do the same and remove this quirk and overhead.

Reply:
The currently commited version is still broken for reverse P/Invokes and it will fail tests. Working on a fix.